src/glossary.ts

```diff
@@ -19,8 +20,14 @@ export interface IsomorphicRequest {
   body?: string
 }
 
-export interface InteractiveIsomorphicRequest extends IsomorphicRequest {
-  respondWith: LazyCallback<(mockedResponse: MockedResponse) => void>
+export class InteractiveIsomorphicRequest extends BufferedRequest {
+  constructor(
+    buffered: BufferedRequest,
+    readonly respondWith: LazyCallback<(mockedResponse: MockedResponse) => void>
```

Comment:
I'd define the `respondWith()` method on the `InteractiveIsomorphicRequest` class straight away. You cannot have an interactive IR without the `respondWith` method. 

The method itself is a deferred method, so it doesn't need to accept anything from the consumer to be defined. The consumer then can await this method being called, and get its return value, by doing `await interactiveRequest.respondWith.invoked()`.

Reply:
also probably should move the `invoked` on IIR as well?

Reply:
No, `invoked()` is a deferred promise from the `respondWith()` method. It doesn't belong to any particular class, only to that lazy function created with `createLazyCallback`. I wouldn't touch this behavior, it works so let's not concern ourselves with it at this moment.

Reply:
I don't understand how respondWith will be defined. Currently we can call `request.respondWith(whatever)`

Reply:
We should define it in the `InteractiveIsomorphicRequest` class:

```ts
class InteractiveIsomorphicRequest {
  public respondWith(response: MockedResponse) {
    return createLazyCallback({ maxCalls: 1, maxCallsCallback, })
  }
}
```

I'd expect this to work unless I'm missing something very important.